### PR TITLE
Migrate from toml to tomllib/tomli

### PR DESCRIPTION
**Pull request by hroncok**
TODO:

 - [ ] Regenerate the poetry lock file somehow (I failed miserably when attempting to do it)
 - [ ] Figure out how to make coverage happy on Python 3.11+ (fasjson_client/config.py lines 32-34)

### CODE REVIEW

File: fasjson_client/config.py [hunk 1]
```diff
@@ -27,7 +27,11 @@
 import os
 from collections.abc import MutableMapping
 
-import toml
+try:
+    import tomllib
+except ImportError:
+    # Python < 3.11
+    import tomli as tomllib
```
Comment (hroncok) [inline fasjson_client/config.py:34]: coverage does not like those 2 lines on Python 3.11

File: fasjson_client/tests/unit/test_config.py [hunk 2]
```diff
@@ -163,8 +170,8 @@ def test_config_load_one_invalid(conf, mocker, tmpdir):
     with pytest.raises(ConfigurationException) as e:
         conf.load_config(config_path)
     expected_msg = (
-        "Failed to parse {}: error at line 1, column 8: Key name found without value. "
-        "Reached end of file."
+        "Failed to parse {}: Expected '=' after a key in a key/value pair "
+        "(at end of document)"
```
Comment (hroncok) [inline fasjson_client/tests/unit/test_config.py:174]: See, the exact error has changed. Is that important for you?

**Comment by hroncok** [comment]
This is as far as I am able to take this. Please help with the rest.

**Comment by abompard** [comment]
I'll take it from here, thanks Miro

**Comment by hroncok** [comment]
I forgot to mention the motivation in the commit message: https://fedoraproject.org/wiki/Changes/DeprecatePythonToml

File: pyproject.toml [hunk 3]
```diff
@@ -39,7 +39,7 @@ requests = "^2.20.0"
 requests-gssapi = "^1.2.1"
 cryptography = {version = ">=2.3", optional = true}
 click = {version = "^6.7 || ^7 || ^8", optional = true}
-toml = "^0.10.0"
+tomli = {version = "*", python = "<3.11"}
```
Comment (hroncok) [inline pyproject.toml:42]: Originally, I had `^2.0` but tomli 2 does not work on Python 3.6 which this supports. The code should work with tomli 1.x as well.

**Pull request merged**